server/src/main/java/org/opensearch/action/admin/cluster/settings/TransportClusterUpdateSettingsAction.java

```diff
@@ -262,13 +268,28 @@ public void onFailure(String source, Exception e) {
 
                 @Override
                 public ClusterState execute(final ClusterState currentState) {
-                    validateCompatibilityModeSettingRequest(request, state);
+                    boolean isCompatibilityModeChanging = validateCompatibilityModeSettingRequest(request, state);
```

Comment:
Just realized that `validateCompatibilityModeSettingRequest` has a bug . If we change just the persistent settings, it doesn't recognize a setting change . We need to check if either of persistent/transient contains the compatibility mode and if the final value is `strict`. Lets modify the IT to just change persistent setting to verify the same as well .

Reply:
We build the `Settings` object by putting the `transient` settings first and then the `persistent` settings like this:

```
Settings settings = Settings.builder().put(request.persistentSettings()).put(request.transientSettings()).build();
```

If the end-user provides different values for the same setting across transient and persistent layers, the transient one takes precedence. The `Settings#put` method is just a Map put under the hood. 

Even if we are setting `null` value for `REMOTE_STORE_COMPATIBILITY_MODE_SETTING`, it gets translated to the default `STRICT` value for the setting, so we would be good there as well.

Reply:
I have modified the new `testIndexSettingsUpdateDuringReplicaCountDecrement` to now use the `putNull` method to reset the settings after migration instead of the `put` method to confirm the `null` check hypothesis